frontend/src/components/form/base/EColorPicker.vue

```diff
@@ -2,69 +2,131 @@
+                :color="serializedValue"
+                class="mt-n1"
+                :aria-label="
+                  pickerOpen
+                    ? $tc('components.form.base.eColorPicker.closePicker')
+                    : $tc('components.form.base.eColorPicker.openPicker', 0, {
+                        label: labelOrEntityFieldLabel,
+                      })
+                "
+                aria-haspopup="true"
+                :aria-expanded="pickerOpen ? 'true' : 'false'"
+                @click="onInputSwatchClick"
+              />
+            </template>
+            <template #append>
+              <slot name="append" />
+            </template>
+          </EColorField>
+        </div>
+      </template>
+      <v-card ref="picker" :ripple="false" tabindex="-1" data-testid="colorpicker">
+        <v-color-picker
+          v-if="pickerNull"
+          key="model"
+          value="#FF0000"
+          :style="{ '--picker-contrast-color': contrast }"
+          flat
+          @update:color="onPickerInput($event.hex)"
+        />
         <v-color-picker
-          :value="removeAlpha(picker.value || '')"
+          v-else
+          key="null"
+          :value="pickerValue"
+          :style="{ '--picker-contrast-color': contrast }"
           flat
-          @input="picker.onInput"
+          @update:color="debouncedPickerValue($event.hex)"
         />
         <v-divider />
         <div class="d-flex gap-2 pa-4 flex-wrap">
           <ColorSwatch
-            v-for="swatch in swatches"
+            v-for="swatch in swatchesWithReset"
             :key="swatch"
             :color="swatch"
-            @select-color="picker.onInput"
+            @select-color="onSwatchSelect($event)"
+          />
+          <ColorSwatch
+            v-if="!required"
+            color="#000000"
+            class="reset"
+            @select-color="onSwatchSelect(null)"
           />
         </div>
       </v-card>
-    </template>
-
-    <!-- passing the append slot through -->
-    <template #append>
-      <slot name="append" />
-    </template>
-  </base-picker>
+    </v-menu>
+  </div>
 </template>
 
 <script>
-import BasePicker from './BasePicker.vue'
 import { formComponentMixin } from '@/mixins/formComponentMixin.js'
 import { contrastColor } from '@/common/helpers/colors.js'
 import ColorSwatch from '@/components/form/base/ColorPicker/ColorSwatch.vue'
+import { debounce } from 'lodash'
+import { formComponentPropsMixin } from '@/mixins/formComponentPropsMixin.js'
 
 export default {
   name: 'EColorPicker',
-  components: { ColorSwatch, BasePicker },
-  mixins: [formComponentMixin],
+  components: { ColorSwatch },
+  mixins: [formComponentMixin, formComponentPropsMixin],
+  inheritAttrs: false,
   props: {
-    value: { type: String, required: true },
+    value: { type: String, required: false, default: null },
   },
+  emits: ['input'],
   data: () => ({
+    pickerOpen: false,
+    pickerValue: null,
+    pickerNull: false,
```

Comment:
for what do you need that?
and couldn't that be a computed?

Reply:
To handle the different problems. If the picker is null, if the picker is open and the not debounced value of the v-colorpicker. Nope, this would create a loop. This is because the v-colorpicker (wrongly) emits if the value is set programatically.

Reply:
ok, thanks